video/decode/vd_lavc.c

```diff
@@ -465,101 +464,110 @@ static void select_and_set_hwdec(struct mp_filter *vd)
 
     m_config_cache_update(ctx->opts_cache);
 
-    bstr opt = bstr0(ctx->opts->hwdec_api);
-
-    bool hwdec_requested = !bstr_equals0(opt, "no");
-    bool hwdec_auto_all = bstr_equals0(opt, "auto") ||
-                          bstr_equals0(opt, "yes") ||
-                          bstr_equals0(opt, "");
-    bool hwdec_auto_safe = bstr_equals0(opt, "auto-safe") ||
-                           bstr_equals0(opt, "auto-copy-safe");
-    bool hwdec_auto_copy = bstr_equals0(opt, "auto-copy") ||
-                           bstr_equals0(opt, "auto-copy-safe");
-    bool hwdec_auto = hwdec_auto_all || hwdec_auto_copy || hwdec_auto_safe;
-
-    if (!hwdec_requested) {
-        MP_VERBOSE(vd, "No hardware decoding requested.\n");
-    } else if (!hwdec_codec_allowed(vd, codec)) {
-        MP_VERBOSE(vd, "Not trying to use hardware decoding: codec %s is not "
-                   "on whitelist.\n", codec);
-    } else {
-        bool hwdec_name_supported = false;  // relevant only if !hwdec_auto
-        struct hwdec_info *hwdecs = NULL;
-        int num_hwdecs = 0;
-        add_all_hwdec_methods(&hwdecs, &num_hwdecs);
-
-        for (int n = 0; n < num_hwdecs; n++) {
-            struct hwdec_info *hwdec = &hwdecs[n];
+    struct hwdec_info *hwdecs = NULL;
+    int num_hwdecs = 0;
+    add_all_hwdec_methods(&hwdecs, &num_hwdecs);
 
-            if (!hwdec_auto && !(bstr_equals0(opt, hwdec->method_name) ||
-                                 bstr_equals0(opt, hwdec->name)))
-                continue;
-            hwdec_name_supported = true;
+    bool hwdec_attempted = false;
+    char *default_hwdec_api[] = {"no", NULL,};
+    char **hwdec_api = ctx->opts->hwdec_api ? ctx->opts->hwdec_api : default_hwdec_api;
+    for (int i = 0; hwdec_api[i]; i++) {
+        bstr opt = bstr0(hwdec_api[i]);
+
+        bool hwdec_requested = !bstr_equals0(opt, "no");
+        bool hwdec_auto_all = bstr_equals0(opt, "auto") ||
+                            bstr_equals0(opt, "yes") ||
+                            bstr_equals0(opt, "");
```

Comment:
not sure why this is even here but equals empty string should be obsolete with this change